src/main/java/kitchenpos/menu/domain/MenuProducts.java

```diff
@@ -0,0 +1,48 @@
+package kitchenpos.menu.domain;
+
+import static kitchenpos.exception.KitchenposExceptionMessage.MENU_PRICE_CANNOT_OVER_THAN_PRODUCT_PRICE;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import kitchenpos.exception.KitchenposException;
+
+@Embeddable
+public class MenuProducts {
+
+    @OneToMany(mappedBy = "menu", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<MenuProduct> menuProducts = new ArrayList<>();
+
+    public void add(List<MenuProduct> menuProducts) {
+        this.menuProducts.addAll(menuProducts);
+    }
+
+    public void connectMenu(final Menu menu) {
+        this.menuProducts.forEach(menuProduct -> menuProduct.connectMenu(menu));
+    }
+
+    public void checkOverPrice(final BigDecimal price) {
+        BigDecimal sum = this.menuProducts.stream()
+                                          .map(this::calculatePrice)
+                                          .reduce(BigDecimal.ZERO, BigDecimal::add);
+        if (price.compareTo(sum) > 0) {
+            throw new KitchenposException(MENU_PRICE_CANNOT_OVER_THAN_PRODUCT_PRICE);
+        }
+    }
+
+    private BigDecimal calculatePrice(MenuProduct menuProduct) {
+        return menuProduct.getProductPrice()
+                          .multiply(BigDecimal.valueOf(menuProduct.getQuantity()));
+    }
```

Comment:
getter 를 사용하기 보다는 객체에 메세지를 보내보면 어떨까요? 😄 
```suggestion
    private BigDecimal calculatePrice(MenuProduct menuProduct) {
        return menuProduct.calculatePrice(menuProduct.getQuantity());
    }
```

Reply:
317a1c1 에서 Price 객체로 로직을 옮겨봤습니다!